deploy/docker-ephemeral/build/Makefile

```diff
@@ -0,0 +1,216 @@
+	docker manifest push $(USERNAME)/$*$(TAGNAME)
+
+#manifest-annotate-<arch>-<name>
+manifest-annotate-%: manifest-create-$$(call snd,%)
+	docker manifest annotate $(USERNAME)/$(call snd,$*)$(TAGNAME) $(USERNAME)/$(call snd,$*)$(TAGNAME)-$(call fst,$*) --arch $(call maniarch,$(call fst,$*)) $(call manivariant,$(call fst,$*))
+
+#manifest-create-<name>
+manifest-create-%: $$(foreach arch,$$(call goodarches,%),upload-$$(arch)-$$*)
+	docker manifest create $(USERNAME)/$*$(TAGNAME) $(patsubst %,$(USERNAME)/$*$(TAGNAME)-%,$(call goodarches,$*)) --amend
+
+# upload-<arch>-<name>
+upload-%: build-%
+	docker push $(USERNAME)/$(call snd,$*)$(TAGNAME)-$(call fst,$*) | cat
+
+# build-<arch>-<name>
+build-%: Dockerfile-$$(foreach target,$$(filter $$(call snd,$$*),$(NOMANIFEST)),NOMANIFEST-)$$* $$(foreach predep,$$(filter $$(call snd,%)-%,$(PREBUILDS)),manifest-push-$$(call snd,$$(predep)))
+	cd $(call snd,$*) && docker build -t $(USERNAME)/$(call snd,$*)$(TAGNAME)-$(call fst,$*) -f Dockerfile-$(call fst,$*) . | cat
+
+
+# with a broken manifest, we have to use a postfix to request docker images other than the one for our native architecture.
+archpostfix=$(foreach arch,$(filter-out $(LOCALARCH),$(1)),-$(arch))
+# with working manifest, we have to use a path when requesting docker images other than the one for our native architecture.
+archpath=$(foreach arch,$(patsubst 386,i386,$(filter-out $(LOCALARCH),$(1))),$(arch)/)
+
+# handle cases where a manifest file is not being respected, and we have to use <name>:<tag>-<arch> format.
+# Dockerfile-NOMANIFEST-<arch>-<name>
+Dockerfile-NOMANIFEST-%: $$(call snd,$$*)/Dockerfile
+	cd $(call snd,$*) && cat Dockerfile | sed "s/^\(MAINTAINER\).*/\1 $(REALNAME) \"$(EMAIL)\"/" | sed "s=^\(FROM \)\(.*\)$$=\1\2$(call archpostfix,$(call fst,$*))=" > Dockerfile-$(call fst,$*)
+
+# handle situations where a manifest is present in upstream, and available as <arch>/<name>:<tag>
+# Dockerfile-<arch>-<name>
+Dockerfile-%: $$(call snd,$$*)/Dockerfile
+	cd $(call snd,$*) && cat Dockerfile | sed "s/^\(MAINTAINER\).*/\1 $(REALNAME) \"$(EMAIL)\"/" | sed "s=^\(FROM \)\(.*\)$$=\1$(call archpath,$(call fst,$*))\2=" > Dockerfile-$(call fst,$*)
+
+# real files, finally!
+
+# define commit IDs for the versions we're using.
+SMTP_COMMIT                = 8ad8b849855be2cb6a11d97d332d27ba3e47483f
+DYNAMODB_COMMIT            = c1eabc28e6d08c91672ff3f1973791bca2e08918
+ELASTICSEARCH_COMMIT       = 06779bd8db7ab81d6706c8ede9981d815e143ea3
+AIRDOCKBASE_COMMIT         = 692625c9da3639129361dc6ec4eacf73f444e98d
+AIRDOCKRVM_COMMIT          = cdc506d68b92fa4ffcc7c32a1fc7560c838b1da9
+AIRDOCKFAKESQS_COMMIT      = 9547ca5e5b6d7c1b79af53e541f8940df09a495d
+JAVAMAVENNODEPYTHON_COMMIT = 645af21162fffd736c93ab0047ae736dc6881959
+LOCALSTACK_COMMIT          = 645af21162fffd736c93ab0047ae736dc6881959
+MINIO_COMMIT               = 118270d76fc90f1e54cd9510cee9688bd717250b
+
+smtp/Dockerfile: 
+	git clone git@github.com:namshi/docker-smtp.git smtp
```

Comment:
For this, and all other `git clone` below: can we use the https urls, rather than the ssh ones? This will fail on CI, as CI otherwise needs to have a ssh key belonging to a github user in order for git clone to work.